### PR TITLE
feat: enable both authentication ways (api key and token)

### PR DESCRIPTION
**Pull request by oscanezing**
enable both authentication ways:
- Api key in header with key Api-Key
- Bearer token

https://merlinapp.atlassian.net/browse/CM-320

### CODE REVIEW

File: authorizer.go [hunk 1]
```diff
@@ -142,6 +143,29 @@ func (a *Authorizer) GinTokenAndBackAuthMiddleware() gin.HandlerFunc {
 	}
 }
 
+func (a *Authorizer) GinTokenAndBackAuthMiddlewareV2() gin.HandlerFunc {
```
Comment (E-ricus) [inline authorizer.go]: Let's create a generic http version for the chi services!
also, maybe TokenAndApiKey as name? we might want on the future, use apikey with front

File: authorizer.go [hunk 2]
```diff
@@ -310,6 +334,17 @@ func readPublicKey(cert []byte) (*rsa.PublicKey, error) {
 	return publicKey, nil
 }
 
+func isApiKey(headers http.Header) bool {
+	apiKey := headers.Get(ApiKeyHeader)
+	if apiKey == "" {
+		return false
+	}
+	if apiKey != os.Getenv("BACKEND_API_KEY") {
```
Comment (E-ricus) [inline authorizer.go]: Don't get the value from env, pass it as a parameter on the middleware

Comment (oscanezing) [reply authorizer.go]: ok, done

File: authorizer.go (hunk 1 above)
Comment (oscanezing) [reply authorizer.go]: ok, done

**Review by mariomurillo — APPROVED** [review]

**Review by E-ricus — APPROVED** [review]

**Pull request merged**